doc/contributor/howto-guide-update-googleapis-sha.md

```diff
@@ -0,0 +1,141 @@
+# How-to Guide: Update googleapis SHA
+
+This document describes the steps required to update the commit SHA for the
+[googleapis][googleapis-repo] repository. The document is intended for
+contributors to the `google-cloud-cpp` libraries. It assumes you are familiar
+with the build systems used in these libraries.
+
+`google-cloud-cpp` depends on the proto files in the
+[googleapis][googleapis-repo]. The build scripts in `google-cloud-cpp` are
+pinned to a specific commit SHA of this repository.  That avoids unexpected
+breakage for us and our customers as `googleapis` makes changes. From time to
+time we need to manually update this commit SHA.  This document describes these
+steps.
+
+## Set your working directory
+
+Go to whatever directory holds your clone of the project, for example:
+
+```shell
+cd $HOME/google-cloud-cpp
+```
+
+Create a branch to make your changes
+
+```shell
+git checkout main
+git checkout -b chore-update-googleapis-sha-circa-$(date +%Y-%m-%d)
+```
+
+## Find out the commit SHA you want to use
+
+Use your browser to find out a recent commit SHA for `googleapis`.  In this
+document we will use `134d8be4b58b2bfc373374f71ec52a0df59af664` as an example.
+That was a recent SHA circa 2022-05-18.
+
+Find the SHA256 of the repository tarball at that commit:
+
+```shell
+SHA=134d8be4b58b2bfc373374f71ec52a0df59af664
+curl -sSL https://github.com/googleapis/googleapis/archive/${SHA}.tar.gz | sha256sum -
+# Output: bf45ef870b4d36de2c7a9eebdd887cd8f9c6c016b65561c93a1f135bb577571a  -
+```
+
+Make a note of the download SHA256.
+
+## Edit the code
+
+You need to edit `cmake/GoogleapisConfig.cmake`
+and `bazel/google_cloud_cpp_deps.bzl`.  You can see sample patches in the
+[appendix](#appendix-sample-patches) at the end of this document.
+
+Commit your edits:
+
+```shell
+git commit -m"chore: update googleapis SHA circa $(date +%Y-%m-%d)" bazel cmake
+```
+
+## Update the generated libraries
+
+```shell
+ci/cloudbuild/build.sh -t generate-libraries-pr
+ ```
+
+## Stage these changes
+
+```shell
+git add .
+```
+
+## Verify everything compiles
+
+```shell
+bazel build //google/cloud/...
+# CMake caches the old protos, clear them before compiling
+rm -fr build-out/*/cmake-out
+ci/cloudbuild/build.sh -t cmake-install-pr
+```
+
+## Commit the generated changes
+
+```shell
+git commit -m"Regenerate and reformat libraries"
```

Comment:
Opt: Perhaps "and reformat" is overkill?

Reply:
Done.  I was also thinking that `git commit --amend --no-edit` would make it easier to create a PR, as the PR title would match the only commit.  Thoughts?